js_modules/dagster-ui/packages/ui-core/src/hooks/useThrottledEffect.tsx

```diff
@@ -0,0 +1,56 @@
+import {DependencyList, useEffect, useRef, useState} from 'react';
+
+/**
+ * A custom hook that throttles the execution of an effect.
+ *
+ * @param {Function} callback - The effect callback to be throttled
+ * @param {Array} dependencies - Array of dependencies for the effect
+ * @param {number} delay - The minimum time (in ms) between effect executions
+ */
+export const useThrottledEffect = (
+  callback: () => void,
```

Comment:
This hook implementation doesn't seem to support the teardown returned by callback(), which it seems like we're expecting to work like useEffect?

Reply:
Good catch 😅